autoapi/templates/python/function.rst

```diff
@@ -1,6 +1,13 @@
 {% if obj.display %}
-.. function:: {{ obj.short_name }}({{ obj.args }}){% if obj.return_annotation is not none %} -> {{ obj.return_annotation }}{% endif %}
+{% for (args, return_annotation) in obj.signatures %}
+{% if loop.index0 == 0 %}
```

Comment:
Would it make more sense to store the overloads in an attribute instead of storing the real function signature with it as well? I feel like the overloads more of a special case of the function.

Reply:
I can’t make a design decision, so I just emulated the Autodoc’s rendering [the figure below].

Python's `@overload` is only for type annotation. I think It's different from the function overloading in languages like C/C++.

<img width="280" alt="sc" src="https://user-images.githubusercontent.com/5351911/90310793-fa0f8700-df2f-11ea-9d6f-08c0ecd80ad5.png">

Alternative approach could be ...

```
.. function:: double(a: int) -> int

.. function:: double(a: str) -> str
   :noindex:

   This is an overloaded function

Reply:
I think the result that you have looks good. What I meant was that we should store overloads as an attribute called "overloads" on the objects, whereas at the moment you have an attribute called signatures that stores all signatures.
So the template would then look like

```
.. function:: {{ obj.short_name }}({{ obj.args }}){% if obj.return_annotation is not none %} -> {{ obj.return_annotation }}{% endif %}

{% for (args, return_annotation) in obj.overloads %}
              {{ obj.short_name }}({{ args }}){% if return_annotation is not none %} -> {{ return_annotation }}{% endif %}

{% endfor %} 
```

Reply:
Okay. I understand.